OSDOCS-7740: updates 4.12.34 rns (#64851)
openshift/openshift-docs: release_notes/ocp-4-12-release-notes.adoc

==== Updating

To update an existing {product-title} 4.12 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster using the CLI].

[id="ocp-4-12-34"]
=== RHBA-2023:5151 - {product-title} 4.12.34 bug fix update

Issued: 2023-09-20

> {product-title} release 4.12.34 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2023:5151[RHBA-2023:5151] advisory. There are no RPM packages for this release.

Reviewer: Obligatory note that this link 404s, but I know that it should be available on release day

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.12.34 --pullspecs
----

[id="ocp-4-12-34-bug-fixes"]
==== Bug fixes

> * Previously, a non-compliant upstream DNS server that provided a UDP response larger than {product-title} specified bufsize of 512 bytes, caused an overflow error in CoreDNS in which a response to a DNS query was not given. With this update, users can configure the `protocolStrategy` field on the `dnses.operator.openshift.io` custom resource to be "TCP". This resolves issues with non-compliant upstream DNS servers. (link:https://issues.redhat.com/browse/OCPBUGS-15251[OCPBUGS-15251])

Reviewer: Shouldn't these OCPBUGS links be in bold? It looks like they were bolded up until 4.12.29, which I'm assuming you copied this from. So maybe consider fixing in both places?

> * Previously, a non-compliant upstream DNS server that provided a UDP response larger than {product-title} specified bufsize of 512 bytes, caused an overflow error in CoreDNS in which a response to a DNS query was not given. With this update, users can configure the `protocolStrategy` field on the `dnses.operator.openshift.io` custom resource to be "TCP". This resolves issues with non-compliant upstream DNS servers. (link:https://issues.redhat.com/browse/OCPBUGS-15251[OCPBUGS-15251])

Reviewer: * Shouldn't these OCPBUGS links be in bold? It looks like they were bolded up until 4.12.29, which I'm assuming you copied this from. So maybe consider fixing in both places?
* The first sentence runs on a bit. I know we don't worry about these bug fix texts as closely as other text, but might be worth adjusting some. I think that the original wording in the release notes text was actually a bit clearer, so I'd recommend going back to some of that. Adding a suggestion, but feel free to ignore or adjust as you see fit

```suggestion
* Previously, a non-compliant upstream DNS server that provided a UDP response larger than the {product-title}-specified `bufsize` of 512 bytes caused CoreDNS to throw an overflow error and not provide a response to a DNS query. With this update, users can configure the `protocolStrategy` field on the `dnses.operator.openshift.io` custom resource to be "TCP". This resolves issues with non-compliant upstream DNS servers. (link:https://issues.redhat.com/browse/OCPBUGS-15251[*OCPBUGS-15251*])
```

> * Previously, the {product-title} Router directed traffic to a route with a weight of `0` when it had only one backend. With this update, the router will not send traffic to routes with a single backend with weight `0`. (link:https://issues.redhat.com/browse/OCPBUGS-18639[OCPBUGS-18639])

Reviewer: "back end" as a noun should be two words per ISG

```suggestion
* Previously, the {product-title} Router directed traffic to a route with a weight of `0` when it had only one back end. With this update, the router will not send traffic to routes with a single back end with weight `0`. (link:https://issues.redhat.com/browse/OCPBUGS-18639[OCPBUGS-18639])
```

> * Previously, the cloud credentials used in Manila CSI Driver Operator were cached, resulting in authentication issues if these credentials were rotated. With this update, this issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-18475[OCPBUGS-18475])

Reviewer: ```suggestion
* Previously, the cloud credentials used in the Manila CSI Driver Operator were cached, resulting in authentication issues if these credentials were rotated. With this update, this issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-18475[OCPBUGS-18475])
```
